File: programovani/ulohy/files/modelovani-obrazcu-i/obdelnik-obrys.cpp
#include <iostream>
using namespace std;


int main() {

    // Deklarace a inicializace    
    const int a = 6; // Inicializace strany obdelnika
    const int b = 4; // Inicializace strany obdelnika

    // Vnejsi cyklus pro radky
    for(int i = 1; i <= b; i++) {

        // Vnitrni cyklus pro jednotlive hvezdicky
        for(int j = 1; j <= a; j++) {            

            // Vypiseme hvezdicky
            if(i == 1 || i == b) {
                cout << "* ";
            }
            else if(j == 1 || j == a) {
                cout << "* ";
            }

            // Pro vnitrek vypiseme pouze mezery
            else {
                cout << "  ";
            }

        }

        // Vypiseme novou radku
        cout << endl;

    }

}

File: programovani/ulohy/files/modelovani-obrazcu-i/obdelnik-obrys.py
# Deklarace a inicializace    
a = 6 # Inicializace strany obdelnika
b = 4 # Inicializace strany obdelnika

# Vnejsi cyklus pro radky
for i in range(1, b + 1):

    for j in range(1, a + 1):   

        # Vypiseme hvezdicky
        if(i == 1 or i == b):
            print("* ", sep="", end="")        
        elif(j == 1 or j == a):
            print("* ", sep="", end="")
        
        # Pro vnitrek vypiseme pouze mezery
        else:
            print("  ", sep="", end="")


    # Vypiseme novou radku
    print(end="\n")